introduction/intro_to_js.md

```diff
@@ -0,0 +1,108 @@
+# Intro to JavaScript world
+
+## Brief history & language development
+
+TODO
+
+## JavaScript basics
+
+### Types
+
+Javascript is dynamically typed language:
+
+```
+var something = 123;
+```
+
+You just don't define types for the variables.
+
+JavaScript is also weakly typed language:
+
+```
+var realNumber = Math.sqrt(2);
+var looksLikeNumber = "123";
+console.log(realNumber * looksLikeNumber); // Outputs "173.9482681718907"
+```
+
+Which means it can perform type coercion implicitly. Without prior asking or failing, as opposed to C#:
+
+```
+double a = 7.3;
+string x = "hi";
+a = a - x; // Fails at compile time with "Operator '-' cannot be applied to operands of type 'double' and 'string'"
+```
+
+Let's look on slightly different example:
+ 
+```
+var doesntlookLikeNumber = "abc123";
+var result = doesntlookLikeNumber * 2;
+console.log(result); // Outputs "NaN" (Not a number)
+console.log(typeof result); // Outputs "number"
+```
+
+Isn't it fun?
+
+### Truthy/Falsy
+ 
+```
+if (!someVariable) { // outputs "Yes" if someVariable is "", 0, NaN, null, undefined or false 
+    console.log("Yes");
+}
+``` 
+
+### Scopes
+
+![Function scope](/assets/img/function_scope.png)
```

Comment:
I would add an example of block scope (let, const, & other languages) as opposed to function scope (var). That can often be a source of confusion and this image does not make it clearer.